### PR TITLE
Add github milestone labeling

### LINKED ISSUE #187: Add milestones
Port the github action workflow from trino and potentially add/update links to the release notes page

### PR DESCRIPTION
**Pull request by mosabua**
This is the same flow as used in the trino repo. Should work the same since the pom setup and such are identical. I am not sure if there is a way to test apart from merging it.

Fixes #187

### CODE REVIEW

File: .github/workflows/milestone.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+name: milestone
+
+on:
+  push:
+    branches: [master]
```
Comment (ebyhr) [inline .github/workflows/milestone.yml:5]: We don't use master branch in this repository.

File: .github/workflows/milestone.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+jobs:
+  set-milestone:
+    if: github.repository_owner == 'trinodb'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Get milestone from pom.xml
+        run: |
+          .github/bin/retry ./mvnw -v
```
Comment (ebyhr) [inline .github/workflows/milestone.yml:20]: retry script doesn't exist in this repository.

**Review by ebyhr — CHANGES_REQUESTED** [review]

File: .github/workflows/milestone.yml (hunk 1 above)
Comment (mosabua) [reply .github/workflows/milestone.yml:5]: dang .. good spotting

File: .github/workflows/milestone.yml (hunk 2 above)
Comment (mosabua) [reply .github/workflows/milestone.yml:20]: will add ..

File: .github/workflows/milestone.yml (hunk 1 above)
Comment (mosabua) [reply .github/workflows/milestone.yml]: Updated to main

**Review by ebyhr — APPROVED** [review]

**Pull request merged**